metaconfig/src/main/scala/org/scalafmt/util/BuildTime.scala

```diff
@@ -0,0 +1,18 @@
+package org.scalafmt.util
+
+import scala.annotation.compileTimeOnly
+import scala.meta._
+
+@compileTimeOnly("Gets expanded away")
+class BuildTime extends scala.annotation.StaticAnnotation {
+
+  inline def apply(defn: Any): Any = meta {
+    defn match {
+      case defn @ Defn.Val(mods, pats, optType, rhs) if pats.size == 1 =>
```

Comment:
I think this might be clearer with a quasiquote, something  in this direction:

```scala
case q"val $name = ???" => q"val $name = ${System.currentTimeMillis()}"
```